frontend/projects/upgrade/src/app/shared-standalone-component-lib/components/common-modal/common-modal.component.ts

```diff
@@ -0,0 +1,44 @@
+import { MatCardModule } from '@angular/material/card';
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+import { FormsModule } from '@angular/forms';
+import { MatButtonModule } from '@angular/material/button';
+import { MatDialogActions, MatDialogClose, MatDialogContent, MatDialogTitle } from '@angular/material/dialog';
+import { CommonModule, NgTemplateOutlet } from '@angular/common';
+import { MatIcon } from '@angular/material/icon';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'app-common-dialog',
+  standalone: true,
+  imports: [
+    MatCardModule,
+    MatFormFieldModule,
+    MatInputModule,
+    FormsModule,
+    MatButtonModule,
+    MatDialogTitle,
+    MatDialogContent,
+    MatDialogActions,
+    MatDialogClose,
+    CommonModule,
+    NgTemplateOutlet,
+    MatIcon,
+  ],
+  templateUrl: './common-modal.component.html',
+  styleUrl: './common-modal.component.scss',
+})
+export class CommonModalComponent {
+  @Input() title = '';
+  @Input() cancelBtnLabel = 'Cancel';
+  @Input() primaryActionBtnLabel = 'Submit';
+  @Input() primaryActionBtnColor = 'primary';
```

Comment:
what if the modal is like an experiment with multiple pages, don't we need an option for the next button as well?

Reply:
the designs for new stuff are not going to do like the experiment design stepper. one the reasons for doing the redesign in the first place was a decision that we don't want to be doing steppers in modals, it's limiting and not the best user experience. these are all the modals that Zack is planning, he intentionally has them all be primary button + close, which is why I decided it was okay to hardcode this design into the component. https://www.figma.com/design/GaKw3nDscTnMEOzIf4CAa7/UpGrade-UI-Latest-14-(Final)?node-id=8283-9405&t=4DR3JobQGZlZ8nxV-0

Reply:
okay, just wanted to confirm it